tensorboard/uploader/uploader.py

```diff
@@ -445,19 +485,146 @@ def _create_point(self, tag_proto, event, value):
+            blob_sequence_id = self._get_or_create_blob_sequence()
+            print("Flushing to blob sequence id: ", blob_sequence_id)
+
+            # TODO(soergel): should we really unpack the tensor here, or ship
+            # it wholesale and unpack server side, or something else?
+            blobs = tensor_util.make_ndarray(self._value.tensor)
+            for seq_index, blob in np.ndenumerate(blobs):
+                # Note the _send_blob() stream is internally rate-limited.
+                self._rpc_rate_limiter.tick()
+                self._send_blob(blob_sequence_id, seq_index[0], blob)
+
+        self._new_request()
+
+    def _get_or_create_blob_sequence(self):
+        request = write_service_pb2.GetOrCreateBlobSequenceRequest(
+            experiment_id=self._experiment_id,
+            run=self._run_name,
+            tag=self._value.tag,
+            step=self._event.step,
+            final_sequence_length=1,
+            metadata=self._metadata,
+        )
+        util.set_timestamp(request.wall_time, self._event.wall_time)
+
+        with _request_logger(request):
+            try:
+                # TODO(@nfelt): execute this RPC asynchronously.
+                response = grpc_util.call_with_retries(
+                    self._api.GetOrCreateBlobSequence, request
+                )
+                blob_sequence_id = response.blob_sequence_id
+            except grpc.RpcError as e:
+                if e.code() == grpc.StatusCode.NOT_FOUND:
+                    raise ExperimentNotFoundError()
+                logger.error("Upload call failed with error %s", e)
+                # TODO(soergel): clean up
+                raise e
+
+        return blob_sequence_id
+
+    def _send_blob(self, blob_sequence_id, seq_index, blob):
+        # TODO(soergel): retry and resume logic
+
+        request_iterator = self._write_blob_request_iterator(
+            blob_sequence_id, seq_index, blob
+        )
+        # TODO(soergel): don't wait for responses for greater throughput
+        # See https://stackoverflow.com/questions/55029342/handling-async-streaming-request-in-grpc-python
+        for response in self._api.WriteBlob(request_iterator):
+            print(repr(response))
+            # TODO(soergel): validate responses?  probably not.
+            pass
+
+    def _write_blob_request_iterator(self, blob_sequence_id, seq_index, blob):
+        # For now all use cases have the blob in memory already.
+        # In the future we may want to stream from disk; that will require
+        # refactoring here.
+        for offset in range(0, len(blob), BLOB_CHUNK_SIZE):
+            chunk = blob[offset : offset + BLOB_CHUNK_SIZE]
+            finalize_object = offset + BLOB_CHUNK_SIZE >= len(blob)
+            request = write_service_pb2.WriteBlobRequest(
+                blob_sequence_id=blob_sequence_id,
+                index=seq_index,
+                data=chunk,
+                offset=offset,
+                crc32c=None,
```

Comment:
Are these TODOs, DO NOT SUBMITs, or nothings?

Reply:
Added TODO.